src/main/java/io/luna/game/model/def/EquipmentDefinition.java

```diff
@@ -225,16 +220,16 @@ public boolean isFullHelmet() {
     }
 
     /**
-     * @return A list of the requirements.
+     * @return An unmodifiable list of the requirements.
      */
-    public ImmutableList<Requirement> getRequirements() {
+    public List<Requirement> getRequirements() {
         return requirements;
     }
 
     /**
-     * @return A list of equipment bonuses.
+     * @return A deep copy of equipment bonuses.
      */
-    public ImmutableList<Integer> getBonuses() {
-        return bonuses;
+    public int[] getBonuses() {
+        return bonuses.clone();
```

Comment:
He's trying to make it so you can't modify the definition. But that's why it used ImmutableList in the first place, I don't really like the idea of having to copy the entire array of bonuses every time

Reply:
Having a copy makes sense here since you want to be able to modify the clone without modifying the parent. I guess my question had more to do with line 165 `bonuses = bonuses.clone()`. More directly, what is the reason for cloning on initialization?

Reply:
I clone on initialization for the same reason that I clone in the getter method; any modifications that are made to the `bonuses` array outside of the `EquipmentDefinition` class should not be reflected within the class. 

I also moved away from using `ImmutableList<Integer>` because of the auto-boxing overhead, but I understand why you wouldn't want to copy the array every time its returned in the getter.

Instead of having a method that returns a copy of the array, we might be able to get away with not exposing the array (or a copy) at all; we can look at what we're currently doing when we return the array, and extract those operations into new methods inside `EquipmentDefinition`.